common/src/api/external/mod.rs

```diff
@@ -200,19 +200,17 @@ where
     }
 }
 
-/// Custom JsonSchema implementation to encode the constraints on Name
-// TODO: 1. make this part of schemars w/ rename and maxlen annotations
-// TODO: 2. integrate the regex with `try_from`
+/// Custom JsonSchema implementation to encode the constraints on Name.
 impl JsonSchema for Name {
     fn schema_name() -> String {
         "Name".to_string()
     }
     fn json_schema(
-        _gen: &mut schemars::gen::SchemaGenerator,
+        _: &mut schemars::gen::SchemaGenerator,
     ) -> schemars::schema::Schema {
-        schemars::schema::Schema::Object(schemars::schema::SchemaObject {
+        schemars::schema::SchemaObject {
             metadata: Some(Box::new(schemars::schema::Metadata {
-                title: Some("A name used in the API".to_string()),
+                title: Some("A contextually unique name".to_string()),
```

Comment:
@davepacheco do you think is a better or worse description?

Reply:
Better!  "contextually" is a little vague though -- maybe "a name unique within the parent collection".  I was going to say "a user-controlled, mutable string identifier that is unique within the object's collection" but maybe those other details are better for the description.

Reply:
I like it; thanks for the suggestion